tests/integration/test_charm.py

```diff
@@ -59,44 +57,22 @@ async def test_logs_write_to_storage(ops_test: OpsTest):
     assert ops_test.model.applications[CHARM_KEY].status == "active"
     assert ops_test.model.applications[DUMMY_NAME].status == "active"
 
-    relation_data = get_provider_data(
-        unit_name=f"{DUMMY_NAME}/0", model_full_name=ops_test.model_full_name
+    produce_and_check_logs(
+        model_full_name=ops_test.model_full_name,
+        kafka_unit_name="kafka/0",
+        provider_unit_name=f"{DUMMY_NAME}/0",
+        topic="hot-topic",
     )
-    logger.debug(f"{relation_data=}")
 
-    topic = "new-topic"
-    username = relation_data.get("username", None)
-    password = relation_data.get("password", None)
-    servers = relation_data.get("uris", "").split(",")
-    security_protocol = "SASL_PLAINTEXT"
 
-    if not (username and password and servers):
-        pytest.fail("missing relation data from app charm")
-
-    client = KafkaClient(
-        servers=servers,
-        username=username,
-        password=password,
-        topic=topic,
-        consumer_group_prefix=None,
-        security_protocol=security_protocol,
+@pytest.mark.abort_on_fail
+async def test_logs_write_to_new_storage(ops_test: OpsTest):
+    check_output(f"JUJU_MODEL={ops_test.model_full_name} juju add-storage kafka/0 log-data")
+    time.sleep(10)  # to give time for storage to complete
```

Comment:
Is there any way in which you can ensure that the storage is attached correctly ?